### ISSUE TITLE
CF error in node-pool with EFS

### ISSUE THREAD

**Issue by mgilbir** [kubernetes-incubator/kube-aws#182]
I'm trying to spin up a node-pool with EFS enabled and I'm getting a CF validation error:

`Error: invalid cloudformation stack: ValidationError: Template format error: Unresolved resource dependencies [SecurityGroupMountTarget] in the Resources block of the template`

When I look into the node-pool rendered stack-template.json there is:
```
    {{if $.ElasticFileSystemID}}
    ,
    "{{$subnetLogicalName}}MountTarget": {
      "Properties" : {
        "FileSystemId": "{{$.ElasticFileSystemID}}",
        "SubnetId": { "Ref": "{{$subnetLogicalName}}" },
        "SecurityGroups": [ { "Ref": "    {{if $.ElasticFileSystemID}}
    ,
    "{{$subnetLogicalName}}MountTarget": {
      "Properties" : {
        "FileSystemId": "{{$.ElasticFileSystemID}}",
        "SubnetId": { "Ref": "{{$subnetLogicalName}}" },
        "SecurityGroups": [ { "Ref": "SecurityGroupMountTarget" } ]
      },
      "Type" : "AWS::EFS::MountTarget"
    }
    {{end}}
" } ]
      },
      "Type" : "AWS::EFS::MountTarget"
    }
    {{end}}
```

with no definition available for the `SecurityGroupMountTarget`

**Comment by mumoshu** [comment]
Thanks for the feedback! Seems to have missed that while implementing node pools.

In the meantime, copy-and-pasting the def for `SecurityGroupMountTarget` from a base cluster's `stack-template.json` into the node pools `stack-template.json` would work.

**Comment by flah00** [comment]
@mumoshu any thoughts on if we can see this fixed in the next rc? Thanks!

**Comment by mumoshu** [comment]
Thanks for bringing this up @flah00 but sorry, no it isn't fixed/planned yet because I'm currently working on a big PR to address another users' feature request.
So, in the meantime, please try the workaround as described in my prev comment and/or submit a PR :bow:

**Comment by flah00** [comment]
Copying and pasting the `SecurityGroupMountTarget` block brought more hassle than it was worth, IMHO. There was a duplication of the target, because the controller and node pools were defining the same resource. Also, management of the worker security group would have to be moved from the main cluster.yaml, which leads to its own set of problems.

Here's what I did to make it work

1. Set `elasticFileSystemId` in `cluster.yaml`
1. `kube-aws up`
1. `kube-aws node-pools init --node-pool-name NAME`
1. Update `node-pools/*/cluster.yaml`
    * Set `workerSecurityGroupIds` to `k8s-sg-worker` security group id
    * Set `kubernetesVersion` to `v1.5.1_coreos.0` (defaults to 1.4.5)
    * Set `elasticFileSystemId`
1. `kube-aws node-pools render stack --node-pool-name NAME`
1. Update `node-pools/*/stack-template.json`
    * Remove the `{{if $.ElasticFileSystemID}}` block
1. You should be able to `kube-aws node-pools up ...`

**Comment by mumoshu** [comment]
Hi @mgilbir and @flah00, long time no see!
FYI, this is fixed since v0.9.4 via #315.
All the necessary security groups created and managed by kube-aws is now imported to each node pool.
You no longer need to modify stack-template.json per node-pool basis.

**Comment by mumoshu** [comment]
I'm closing this as resolved but please feel free to reopen if necessary. Thanks for your patience 🍵
